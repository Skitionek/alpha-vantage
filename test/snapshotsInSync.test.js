/* DOCUMENT INFORMATION
	- Author:   Dominik Maszczyk
	- Email:    Skitionek@gmail.com
	- Created:  23/04/2019
*/

import { obtainStructure } from "./jest.extensions";
import { integrationSnapshots, unitSnapshots } from "./dataStructure";

describe.each(Object.keys(integrationSnapshots))("%s", groupKey => {
	const integrationSnapshot = integrationSnapshots[groupKey];
	const unitSnapshot = unitSnapshots[groupKey];
	it('contains same groups', ()=>
		expect(Object.keys(unitSnapshot))
			.toEqual(Object.keys(integrationSnapshot)));
	Object.keys(integrationSnapshot).forEach(key => {
		it(key, () => {
			expect(obtainStructure(unitSnapshot[key])).toEqual(integrationSnapshot[key]);
		});
	})
});

