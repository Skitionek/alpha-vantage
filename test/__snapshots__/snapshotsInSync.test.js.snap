// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crypto daily 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`crypto exchangeRates 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`crypto exchangeTimeSeries 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`crypto monthly 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`crypto weekly 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data daily 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data daily_adjusted 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data exchangeTimeSeries 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data exchangeTimeSeries_adjusted 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data intraday 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data monthly 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data monthly_adjusted 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data quote 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data search 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data weekly 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`data weekly_adjusted 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`forex daily 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`forex exchangeRates 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`forex exchangeTimeSeries 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`forex intraday 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`forex monthly 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`forex weekly 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`performance sector 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ad 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical adosc 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical adx 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical adxr 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical apo 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical aroon 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical aroonosc 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical atr 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical bbands 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical bop 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical cci 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical cmo 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical dema 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical dx 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ema 1`] = `
Snapshot Diff:
- First value
+ Second value

  Object {
    "data": Array [
      Object {
+       "close": "Float",
        "date": "Timestamp/Date",
+       "high": "Float",
+       "low": "Float",
+       "open": "Float",
        "value": "Float",
      },
    ],
+   "from_symbol": "String",
    "indicator": "String",
+   "information": "String",
    "interval": "Interval",
    "last_refreshed": "Timestamp/Date",
+   "output_size": "String",
    "series_type": "String",
    "symbol": "String",
    "time_period": "Float",
    "time_zone": "String",
+   "to_symbol": "String",
  }
`;

exports[`technical ht_dcperiod 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ht_dcphase 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ht_phasor 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ht_sine 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ht_trendline 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ht_trendmode 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical kama 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical macd 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical macdext 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical mama 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical mfi 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical midpoint 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical midprice 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical minus_di 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical minus_dm 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical mom 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical natr 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical obv 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical plus_di 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical plus_dm 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ppo 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical roc 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical rocr 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical rsi 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical sar 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical sma 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical stoch 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical stochf 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical stochrsi 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical t3 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical tema 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical trange 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical trima 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical trix 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical ultosc 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical vwap 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical willr 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`technical wma 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
